cluster/schema/schema.go

```diff
@@ -210,11 +212,12 @@ type shardReader interface {
 	GetShardsStatus(class, tenant string) (models.ShardStatusList, error)
 }
 
-func NewSchema(nodeID string, shardReader shardReader) *schema {
+func NewSchema(nodeID string, shardReader shardReader, classTenantDataEvents chan metadataserver.ClassTenant) *schema {
```

Comment:
Same here, may be another constructor something like `NewSchemaWithTenantEvents`